gazelle/python/generate.go

```diff
@@ -463,6 +470,20 @@ func hasEntrypointFile(dir string) bool {
 	return false
 }
 
+// hasNameEqualsMain determines if the file contains 'if __name__ == "__main__"'.
+func hasNameEqualsMain(path string) bool {
+	searchString := `if __name__ == ['"]__main__['"]:`
+	bytesContents, err := os.ReadFile(path)
+	if err != nil {
+		return false
+	}
+	match, err := regexp.Match(searchString, bytesContents)
```

Comment:
Reading from the end of the file line by line in reverse and matching the whole line (stripped) could be faster? If you want to still use `regexp`, you could at least compile the regexp upfront and store it as a `var` at the top of the file.

Reply:
Yeah, let me look into making this faster.